src/app/shared/change-password-dialog/password-check.service.ts

```diff
@@ -0,0 +1,36 @@
+/** Angular Imports */
+import { Injectable } from '@angular/core';
+import { ValidatorFn, AbstractControl, ValidationErrors } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+/**
+ * Password Check service.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class PasswordCheckService {
+
+  constructor() { }
+
+  /**
+   * Confirm Change Password of Users
+   * @param controlNameToCompare Form Control Name to be compared.
+   */
+  confirmPassword(controlNameToCompare: string): ValidatorFn {
+    return (c: AbstractControl): ValidationErrors|null => {
+        if (c.value == null || c.value.length === 0) {
+            return null;
+        }
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
```

Comment:
can you tell me what did you refer to while making this function

Reply:
Referred to a stack overflow solution. I edited this according to my problem

Reply:
can you find the link please?

Reply:
https://stackoverflow.com/questions/51605737/confirm-password-validation-in-angular-6

Scroll down. You'll find a similar solution for this. I can't remember the exact link as I found this solution long time ago.

Reply:
Is this okay?

Reply:
Ok